tools/symbol-check.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python3
+# Copyright (c) 2014 Wladimir J. van der Laan
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+'''
+A script to check that a secp256k1 shared library
+exports only expected symbols.
+
+Example usage:
+
+    ./tools/symbol-check.py .libs/libsecp256k1.so.0.0.0
+or
+    ./tools/symbol-check.py .libs/libsecp256k1-0.dll
+or
+    ./tools/symbol-check.py .libs/libsecp256k1.0.dylib
+'''
+import re
+import sys
+import subprocess
+from typing import List
+
+import lief #type:ignore
+
+def grep_exported_symbols() -> List[str]:
+    grep_output = subprocess.check_output(["git", "grep", "^SECP256K1_API", "--", "include"], universal_newlines=True, encoding="utf8")
+    lines = grep_output.split("\n")
+    exports: List[str] = []
+    for line in lines:
+        if line.strip():
+            function_name = re.sub(r'\W', '', line.split(" ")[-1])
```

Comment:
this doesn't work with `include/secp256k1.h:SECP256K1_API void secp256k1_selftest(void);` for which the script assumes the symbol is `secp256k1_selftestvoid`

Reply:
Thanks! [Fixed](https://github.com/bitcoin-core/secp256k1/pull/1135#issuecomment-1348812918).